internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -439,9 +527,124 @@ func validateSecretConfig(secretConfig SecretConfig) []string {
 		missingKeys = append(missingKeys, "type")
 	}
 
-	if secretConfig.IntegrationName == "" {
-		missingKeys = append(missingKeys, "integration_name")
+	return missingKeys
+}
+
+var availableRotatingSecretProviders = map[string]map[string]any{
+	string(integrations.Twilio):       twilioRotatingSecretConfig,
+	string(integrations.MongoDBAtlas): mongoDBAtlasRotatingSecretConfig,
+	string(integrations.AWS):          awsRotatingSecretConfig,
+	string(integrations.GCP):          gcpRotatingSecretConfig,
+}
+
+var availableDynamicSecretProviders = map[string]map[string]any{
+	string(integrations.AWS): awsDynamicSecretConfig,
+	string(integrations.GCP): gcpDynamicSecretConfig,
+}
+
+func promptUserForConfig(opts *CreateOpts) (SecretConfig, secretConfigInternal, error) {
+	var (
+		config         SecretConfig
+		internalConfig secretConfigInternal
+		err            error
+		providerFields map[string]map[string]any
+		items          []string
+	)
+
+	if !opts.IO.CanPrompt() {
+		return config, internalConfig, fmt.Errorf("unable to create secret interactively")
 	}
 
-	return missingKeys
+	if opts.Type == secretTypeDynamic {
+		providerFields = availableDynamicSecretProviders
+		items = maps.Keys(providerFields)
+	} else if opts.Type == secretTypeRotating {
+		providerFields = availableRotatingSecretProviders
+		items = maps.Keys(providerFields)
+	}
+
+	providerPrompt := promptui.Select{
+		Label:  "Please select the provider you would like to configure",
+		Items:  items,
+		Stdin:  io.NopCloser(opts.IO.In()),
+		Stdout: iostreams.NopWriteCloser(opts.IO.Err()),
+	}
+
+	_, provider, err := providerPrompt.Run()
+	if err != nil {
+		return config, internalConfig, fmt.Errorf("provider selection prompt failed: %w", err)
+	}
+	config.Type = integrations.IntegrationType(provider)
+
+	fieldValues, err := populateFieldValues(providerFields[provider], opts)
```

Comment:
Is the provider here guaranteed to be a valid? (otherwise `providerFields[provider]` might return nil)

Reply:
Yep, it's guaranteed to be valid. The provider was chosen by the user from a selection of providers in `availableDynamicSecretProviders` and `availableRotatingSecretProviders` ([maps](https://github.com/hashicorp/hcp/pull/167/files#diff-7f92b17a95650abeca3ec432e8bcf6859f19709d1bdb6b3687fb5d704940a4ecR533-R543) from provider to fields)